epistasis_selection.py

```diff
@@ -74,18 +90,22 @@ def get_epistatic_seqs_for_landscape(landscape: potts_model.PottsModel,
   if distance % 2 != 0:
     raise ValueError('Odd distance not supported.')
 
-  # TODO(nthomas) another option is to do this combination in batches, until the test set is full or the
-  # input mutations are exhausted
   if not top_k:
     top_k = n
   mutation_pairs = utils.get_top_n_mutation_pairs(landscape.epistasis_tensor, top_k, lowest=not adaptive)
+  if max_reuse is not None:
+    assert max_reuse > 0
+    mutation_pairs = filter_mutation_set_by_position(mutation_pairs, limit=max_reuse)
```

Comment:
note that this function is greedy and get_top_n_mutation_pairs also has a distinct ordering to its outputs. Maybe randomize mutation_pairs before filtering?

Reply:
Yeah I think we should note that this function is greedy in the docstring. I think it's a feature that we want to keep. We want the resulting set of epistatic sequences to be enriched for the highest magnitude epistatic interactions possible, so I think the greedy filtering makes sense.